### DISCUSSION TITLE
[HW4] Can I create own module in synthesize.ml?

### DISCUSSION THREAD

**Discussion by Akais24** [prosyslab-classroom/cs424-program-reasoning#438] [Q&A] [answered]
### Your Name

황재영

### Question

HW4 과제에서
synthesis.ml에서 기존 모듈 바깥에, 새로운 모듈이나 함수 등을 추가해도 괜찮을까요?

---

In the HW4 assignment, is it okay to add new modules or functions outside of the existing modules in synthesis.ml?

**Comment by spearo2** [comment, marked as answer]
과제의 README.md에 다음과 같이 설명되어 있습니다.
```
Notice that students are not permitted to change directory structures and types of the functions. 
All the functions you implement must have the same types as described in the module signatures. 
However, students are allowed to change let to let rec if needed.
```
재영님이 말씀하시는 새로운 모듈이나 함수등의 추가가 위의 규칙을 만족한다면 괜찮습니다.
특별히 프로그램을 합성하는 모든 주요 함수와 모듈의 처음 주어진 signature와 타입을 만족해야 합니다.

==========

The instructions in the README.md file of the homework state that
```
Notice that students are not permitted to change directory structures and types of functions. 
All the functions you implement must have the same types as described in the module signatures. 
However, students are allowed to change let to let rec if needed.
```
It should be okay if the modules or functions you mentioned follow the given rule.
All the major modules and functions that are given must keep their signatures and types as they were.
